src/common/fldout_nc.f90

```diff
@@ -1425,33 +1431,55 @@ subroutine accumulate_fields(tf1, tf2, tnow, tstep, nsteph)
   end do
 
 !..accumulated/integrated concentration
+  block
+  use snapfldML, only: sigma_level_of_surface_h => field1, surface_temp => field2, &
+                       surface_pressure =>field3, dh=>field4, t1, t2
+  use snaptabML, only: hypsometric_eq_inv
+  integer :: ilvl
 
-  concen = 0.0
+  if (surface_layer_is_lowest_level) then
+    do n=1,npart
+      part = pdata(n)
+      ilvl = part%z*10000.0
+      k = ivlayer(ilvl)
+      if (k==1) then
+        i = hres_pos(part%x)
+        j = hres_pos(part%y)
+        m = def_comp(part%icomp)%to_running
+        concen(i,j,m) = concen(i,j,m) + dble(part%rad())
+      endif
+    enddo
+    dh(:,:) = rt1*hlayer1(:,:,1) + rt2*hlayer2(:,:,1)
+  else
+    ! Use a predetermined height as the surface layer
+    ! Approximate temperature by using first model layer
+    surface_temp(:,:) = rt1*t1(:,:,2) + rt2*t2(:,:,2)
+    surface_pressure(:,:) = rt1*ps1 + rt2*ps2
+    ! pressure at 30m height (negative depth)
+    sigma_level_of_surface_h(:,:) = hypsometric_eq_inv(-surface_height_m, surface_pressure, surface_temp)
+    sigma_level_of_surface_h(:,:) = sigma_level_of_surface_h / surface_pressure
+
```

Comment:
maybe sigma_level_of_surface_h should be generated on reading meteo rather than on writing, and then interpolate here for two timesteps? Or, if this is slow/memory intensive, use a simple tabulation: total variation is about 10%, so 1% accuracy could be easily by an small table.